node_modules/gh-groups/lib/authz.js

```diff
@@ -59,3 +59,49 @@ var canUpdateGroupMembers = module.exports.canUpdateGroupMembers = function(ctx,
     // retrieve the members of a group, you are also allowed to update them
     canGetGroupMembers(ctx, group, callback);
 };
+
+/**
+ * Determine whether or not the user in context can lock a group
+ *
+ * @param  {Context}    ctx                         Standard context object containing the current user and the current application
+ * @param  {Group}      group                       The group to lock
+ * @param  {Function}   callback                    Standard callback function
+ * @param  {Object}     callback.err                An error that occurred, if any
+ * @param  {Boolean}    callback.canLockGroup       `true` if the user in context has the appropriate permission. `false` otherwise
+ */
+var canLockGroup = module.exports.canLockGroup = function(ctx, group, callback) {
+    // Ensure the current user can lock the group. If you can
+    // retrieve the members of a group, you are also allowed to lock it
+    canGetGroupMembers(ctx, group, function(err, canGetMembers) {
+        if (err || !canGetMembers) {
+            return callback(err, canGetMembers);
+        }
+
+        // We also need to check the lock itself
+        var isLockedForUser = (group.isLocked() && group.LockedBy && (group.LockedBy.id !== ctx.user.id && !ctx.user.canAdmin(group.AppId)));
```

Comment:
I'm a bit hesitant to move that logic to the DB as we'd then also have to do the isMember check. The `canReleaseLock` call could just call the `canLockGroup` function to remove the redundant code.